src/modules/bunnymq.js

```diff
@@ -0,0 +1,347 @@
+      callback = options;
+      // default message options
+      options = { persistent: true, durable: true };
+    }
+
+    // consumer gets a suffix if one is set on the configuration, to suffix all queues names
+    // ex: service-something with suffix :ci becomes service-suffix:ci etc.
+    const suffixedQueue = queue + this._connection.config.consumerSuffix;
+
+    return this._connection.get()
+    .then((channel) => {
+      this.channel = channel;
+
+      // when channel is closed, we want to be sure we recreate the queue ASAP so we trigger a reconnect by recreating the consumer
+      this.channel.addListener('close', () => {
+        this.consume(queue, options, callback);
+      });
+
+      return this.channel.assertQueue(suffixedQueue, options)
+      .then((q) => {
+        this._connection.config.transport.info('bmq:consumer', 'init', q.queue);
+
+        this.channel.consume(q.queue, (msg) => {
+          this._connection.config.transport.info('bmq:consumer', `[${q.queue}] < ${msg.content.toString()}`);
+
+          // main answer management chaining
+          // receive message, parse it, execute callback, check if should answer, ack/reject message
+          Promise.resolve(parsers.in(msg))
+          .then(callback)
+          .then(this.checkRpc(msg, q.queue))
+          .then(() => {
+            this.channel.ack(msg);
+          })
+          .catch((err) => {
+            // if something bad happened in the callback, reject the message so we can requeue it (or not)
+            this._connection.config.transport.error('bmq:consumer', err);
+            this.channel.reject(msg, this._connection.config.requeue);
+          });
+        }, { noAck: false });
+
+        return true;
+      });
+    })
+    .catch(() =>
+      // in case of any error creating the channel, wait for some time and then try to reconnect again (to avoid overflow)
+      utils.timeoutPromise(this._connection.config.timeout)
+        .then(() => this.subscribe(queue, options, callback))
+    );
+  }
+}
+
+class BunnyMQ {
+  constructor(connection) {
+    this._connection = connection;
+    this.producer = new Producer(connection);
+    this.consumer = new Consumer(connection);
+  }
+
+  get connection() {
+    return this._connection;
+  }
+
+  set connection(value) {
+    this._connection = value;
+    this.producer.connection = value;
+    this.consumer.connection = value;
+  }
+
+  // for backward compatibility. @deprecated
+  consume(queue, options, callback) {
+    return this.consumer.subscribe(queue, options, callback);
+  }
+
+  subscribe(queue, options, callback) {
+    return this.consumer.subscribe(queue, options, callback);
+  }
+
+  // for backward compatibility. @deprecated
+  produce(queue, msg, options) {
+    return this.producer.publish(queue, msg, options);
+  }
+
+  publish(queue, msg, options) {
+    return this.producer.publish(queue, msg, options);
+  }
+}
+
+let instance;
+module.exports = (connection) => {
```

Comment:
would it make more sense to export a default instance that is also configured like this but also allow users to create instances as they see fit? maybe export the classes also. Just food for thought